lib/Base/IO.cpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Base/IO.h"
+
+#include <cstdio>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+namespace {
+
+static size_t filesize(std::string filename) {
```

Comment:
const ref.

Reply:
+1 on StringRef. 

I think that LLVM has portable utilities that implement this but also compile on windows.

Reply:
I didn't want to use StringRef because it doesn't guarantee a null terminator :-/